src/api/index.js

```diff
@@ -0,0 +1,272 @@
+import config from '../config'
+import {mergeRecursive} from "../helpers/mergeObjects";
+import redirect from "react-router/es/Redirect";
+
+
+class ApiClass {
+    constructor() {
+        this.url = `${config.apiUrl}`;
+    }
+
+    makeRequest = ({url, data, token}) => {
+        const _data = {
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json',
+                ...!!token && {Authorization: `Bearer ${token}`},
+
+            },
+            credentials: 'include',
+
+        };
+
+        const mergedData = mergeRecursive(data, _data);
+
+        return fetch(url, mergedData)
+            .then(response => {
+                if (response.status <= 300) {
+                    return response;
+                }
+                console.log(response.status)
+                if (response.status === 302) {
+                    redirect(response.headers.location);
+                }
+
+                return response.json().then(data => {
+                    if (response.status >= 300) {
+                        return data.message
+                    }
+                    return data;
+                });
+            })
+            .catch(e => Error(e));
+
+    };
+
+    fetchCreateUser = user => {
+        return this.makeRequest({
+            url: `${this.url}/signUp`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    username: user.username,
+                    password: user.password,
+                    email: user.email
+                })
+            }
+        })
+    };
+
+
+    fetchLoginUser = user => {
+        return this.makeRequest({
+            url: `${this.url}/login`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    email: user.email,
+                    password: user.password
+                })
+            }
+        })
+    };
+
+    fetchLoginWithGoogleUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/login/google`,
+            data: {
+                method: 'GET',
+
+            }
+        })
+    };
+
+    fetchGetTokenUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/getToken`,
+            data: {
+                method: 'GET',
+
+            }
+        })
+    };
+
+    fetchArtists = () => {
+        return this.makeRequest({
+            url: `${this.url}/artists`,
+            data: {
+                method: 'GET',
+            },
+            token: localStorage.token
+        })
+    };
+
+    fetchAddToDict = (WordId, image, songId) => {
```

Comment:
what do you fetch here? 
It seems that you adding item to dict here but not fetching here?
for me this name is little bit misleading.
What do you think about this?

Reply:
what about same naming convention
ex. wordId, songId.
or there is a reason of different naming conventions?